module_info_import/models/module_information.py

```diff
@@ -1,107 +1,92 @@
-                "Error module sync API : %s : %s",
-                data.get("name"),
-                data.get("description"),
-            )
-            raise UserError(ERROR_MESSAGE)
-        return data
+    def synchronize_module(self):
+        for version in VERSIONS:
+            data = self.get_module_info(version)
+            for orga, repos in data.items():
+                for repo, modules in repos.items():
+                    for module in modules:
+                        self._update_or_create_modules(version, orga, repo, module)
 
     @api.model
-    def _get_module_information_vals_from_github_stat(self, module_info):
-        return {
-            "name": module_info["name"],
-            "description_rst": module_info["description_rst"],
-            "description_html": module_info["description_rst_html"],
-            "authors": module_info["author_ids_description"],
+    def _update_or_create_modules(self, version, orga_name, repo_name, module):
+        odoo_version_dct = {
+            v.version: v.id for v in self.env["odoo.version"].search([])
+        }
+        repo = self._get_or_create_repo(orga_name, repo_name)
+        version_id = odoo_version_dct.get(version, "odoo_version_unknown")
+        vals = {
+            "available_version_ids": [(4, version_id, 0)],
+            "repo_id": repo.id,
+            "technical_name": module,
         }
+        module = self.search(
+            [("technical_name", "=", module), ("partner_id", "=", False)]
+        )
+        if module:
+            if module._should_update_module(version, orga_name):
+                module.write(vals)
+            module._add_available_version(version)
+        else:
+            module = self.create(vals)
 
     @api.model
-    def _update_modules(self, data):
-        # Since the goal is to resynchronize all modules and versions, it
-        # seems appropriate to make one big search and then work with cache.
-        #        existing_module = self.search([])
-        #        test = dict([(x.name, x.id) for x in existing_module])
-        #        existing_module_names = test.keys()
-        #        for module_info in data:
-        #            if module_info['technical_name'] in existing_module_names:
-        #                module_id = test.get(module_info['technical_name'])
-        #                self.browse(module_id).write({'name': module_info['technical_name']})
-        #            else:
-        #                self.create({'name': {'name': module_info['technical_name']}})
-        odoo_version_dct = dict(
-            [(v.version, v.id) for v in self.env["odoo.version"].search([])]
+    def _get_or_create_repo(self, orga_name, repo_name):
+        repo_vals = {"organization": orga_name, "name": repo_name}
+        repo = self.env["module.repo"].search(
+            [("name", "=", repo_name), ("organization", "=", orga_name)], limit=1
         )
+        if repo and repo.organization != orga_name:
```

Comment:
simplify this code
Testing "repo.organization != orga_name" is useless as we search on the organization just before